simde/simde-common.h

```diff
@@ -1034,6 +1091,9 @@ HEDLEY_DIAGNOSTIC_POP
 #      if !SIMDE_DETECT_CLANG_VERSION_CHECK(11,0,0)
 #        define SIMDE_BUG_CLANG_BAD_VGET_SET_LANE_TYPES
 #      endif
+#      if defined(SIMDE_ARM_NEON_A32V7_NATIVE) && !defined(SIMDE_ARM_NEON_A32V8_NATIVE)
+#        define SIMDE_BUG_CLANG_A32V7_MISSFUNC
```

Comment:
Please file a bug with llvm and change this to refer to that bug number

Reply:
This error is not due to a missing function, and I have changed this error to `SIMDE_BUG_CLANG_QEMU_BUS_ERROR_7`.

When I use the cross-file `armv7-clang-16-ccache.cross` to build the SIMDe test suite, it can be successfully built. However, it encounters the error `Bus error (core dumped)` when run using QEMU (`qemu-arm-static -L /usr/arm-linux-gnueabihf`). Where should I report this problem, CLANG or QEMU?

Reply:
Bus errors are typically due to unaligned memory accesses. 

In the pseudo code https://developer.arm.com/architectures/instruction-sets/intrinsics/vld2q_dup_p16 there is a reference to `CheckSPAlignment`. Obviously it would have been nicer if the documentation was more explicit.

You can fix this by modifying the test code to align the input data. Look for the SIMDE_ALIGN family of macros.

Reply:
In the function `vld2q_dup_p16`, it ran successfully after I added `SIMDE_ALIGN_TO_16` to each variable. However, the arrangement of results is not the expected one. 
The expected results are as follows:
```
{ { SIMDE_POLY16_C(23152), SIMDE_POLY16_C( 8638) },
        { { SIMDE_POLY16_C(23152), SIMDE_POLY16_C(23152), SIMDE_POLY16_C(23152), SIMDE_POLY16_C(23152), SIMDE_POLY16_C(23152), SIMDE_POLY16_C(23152), SIMDE_POLY16_C(23152), SIMDE_POLY16_C(23152) },
          { SIMDE_POLY16_C( 8638), SIMDE_POLY16_C( 8638), SIMDE_POLY16_C( 8638), SIMDE_POLY16_C( 8638), SIMDE_POLY16_C( 8638), SIMDE_POLY16_C( 8638), SIMDE_POLY16_C( 8638), SIMDE_POLY16_C( 8638) } },
    },
```
These results can be generated by `armv7-gcc-12-ccache.cross` or `aarch64-clang-15-ccache.cross`.

---
However, the results generated by the cross-file `armv7-clang-16-ccache.cross` are as follows:
```
{ { SIMDE_POLY16_C(44945), SIMDE_POLY16_C(63248) },
        { { SIMDE_POLY16_C(63248), SIMDE_POLY16_C(63248), SIMDE_POLY16_C(63248), SIMDE_POLY16_C(63248), SIMDE_POLY16_C(44945), SIMDE_POLY16_C(44945), SIMDE_POLY16_C(44945), SIMDE_POLY16_C(44945) },
          { SIMDE_POLY16_C(    0), SIMDE_POLY16_C(    0), SIMDE_POLY16_C(    0), SIMDE_POLY16_C(    0), SIMDE_POLY16_C(63248), SIMDE_POLY16_C(63248), SIMDE_POLY16_C(63248), SIMDE_POLY16_C(63248) } },
    },
```

Could there be some bugs in CLANG when using the `-march=armv7-a` flag?

Reply:
> These results can be generated by `armv7-gcc-12-ccache.cross` or `aarch64-clang-15-ccache.cross`.

As a rule, we only generate test data using real hardware and the actual intrinsic. The one exception to the real hardware requirement being a vendor provided simulator like SDE from Intel.

If you used `armv7-gcc-12-ccache.cross` then that was using the plain implementation not the real intrinsic due to `SIMDE_BUG_GCC_95399`

https://godbolt.org/z/o53PhPoar

> Could there be some bugs in CLANG when using the `-march=armv7-a` flag?

Yes, seems so. See the results from the above Compiler Explorer link.

Reply:
> If you used `armv7-gcc-12-ccache.cross` then that was using the plain implementation not the real intrinsic due to `SIMDE_BUG_GCC_95399`

I did not notice that the GCC bug macro blocked the native implementation. You are right.

> https://godbolt.org/z/o53PhPoar
> 
> > Could there be some bugs in CLANG when using the `-march=armv7-a` flag?
> 
> Yes, seems so. See the results from the above Compiler Explorer link.

https://github.com/llvm/llvm-project/issues/71763